src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -1245,25 +1249,37 @@ void TaskSketcherElements::changeEvent(QEvent *e)
 /* Settings menu ==================================================*/
 void TaskSketcherElements::createSettingsButtonActions()
 {
-    QAction* action = new QAction(tr("Extended information"), this);
+    QAction* action = new QAction(tr("Snap to objects"), this);
+    QAction* action2 = new QAction(tr("Extended information (in widget)"), this);
 
     action->setCheckable(true);
+    action2->setCheckable(true);
 
-    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/Elements");
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher");
     {
         QSignalBlocker block(this);
-        action->setChecked(hGrp->GetBool("ExtendedNaming", false));
+        action->setChecked(hGrp->GetGroup("General")->GetBool("SnapToObjects", true));
+        action2->setChecked(hGrp->GetGroup("Elements")->GetBool("ExtendedNaming", false));
     }
 
     ui->settingsButton->addAction(action);
+    ui->settingsButton->addAction(action2);
 
     isNamingBoxChecked = hGrp->GetBool("ExtendedNaming", false);
 }
 
+void TaskSketcherElements::onSettingsSnapToObjectChanged()
+{
+    QList<QAction*> acts = ui->settingsButton->actions();
+
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General");
+    hGrp->SetBool("SnapToObjects", acts[0]->isChecked());

```

Comment:
Better to use same code structure as function just below.

Reply:
What do you mean?
`isNamingBoxChecked = acts[1]->isChecked();` is not needed because the value is not used anywhere in the widget.

`slotElementsChanged();` is not needed either.

Reply:
There are 2 functions that do same thing. Better if they do it the same way, whatever it is.